### PR TITLE
Add GitHub actions CI

### PR DESCRIPTION
**Pull request by jamessan**
Example run: https://github.com/jamessan/pynvim/actions/runs/382403759

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+      - if: matrix.target == 'checkqa'
+        run: printf 'TOXENV=checkqa,docs\n' >> $GITHUB_ENV
+
+      - if: matrix.target == 'tests'
+        run: |
+          mkdir "$GITHUB_WORKSPACE/_neovim"
+          wget -q -O - https://github.com/neovim/neovim/releases/download/nightly/nvim-linux64.tar.gz | tar xzf - --strip-components=1 -C "$GITHUB_WORKSPACE/_neovim"
+          printf 'PATH=%s:%s\n' "$GITHUB_WORKSPACE/_neovim/bin" "$PATH" | tee -a $GITHUB_ENV
+          printf 'VIMRUNTIME=%s\n' "$GITHUB_WORKSPACE/_neovim/share/nvim/runtime" | tee -a $GITHUB_ENV
```
Comment (jamessan) [inline .github/workflows/ci.yml:34]: This type of thing might be worth creating an actual GitHub action (e.g., neovim/action) for people to use.

Comment (bfredl) [reply .github/workflows/ci.yml:34]: yes :)

**Comment by justinmk** [comment]
Rebased. (Previous: https://github.com/neovim/pynvim/pull/469/commits/85d3756e2ffcb11ee492273a2e29e143b005acb1 )

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -59,33 +72,14 @@ jobs:
...
       run: |
         echo $PATH
         which nvim
         nvim --version
-        python -m pytest
-
-    - uses: codecov/codecov-action@v3
-      if: runner.os == 'macOS'
-      with:
-        verbose: true # optional (default = false)
```
Comment (justinmk) [inline .github/workflows/test.yml:75]: I gave up on codecov for now. https://github.com/neovim/pynvim/pull/469/commits/68bde2069b58b369a2a87ecdade8e77d28dbe68d

```
    ERROR: Failed building wheel for pyuv
    ERROR: Could not build wheels for pyuv, which is required to install pyproject.toml-based projects
```

File: tox.ini [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
 deps =
   pytest-timeout
-  cov: pytest-cov
-  pyuv: pyuv
-setenv =
-  cov: PYTEST_ADDOPTS=--cov=. {env:PYTEST_ADDOPTS:}
+# cov: pytest-cov
+# pyuv: pyuv
+# setenv =
+#   cov: PYTEST_ADDOPTS=--cov=. {env:PYTEST_ADDOPTS:}
```
Comment (justinmk) [inline tox.ini:19]: I gave up on codecov for now. https://github.com/neovim/pynvim/pull/469/commits/68bde2069b58b369a2a87ecdade8e77d28dbe68d

```
    ERROR: Failed building wheel for pyuv
    ERROR: Could not build wheels for pyuv, which is required to install pyproject.toml-based projects
```

**Comment by wookayin** [comment]
By the why are we using tox? rather than just running pytest, linters directly?  etc. I find tox makes debugging CI much harder.

**Comment by justinmk** [comment]
> By the why are we using tox? rather than just running pytest, linters directly? etc. I find tox makes debugging CI much harder.

No objection to removing it.

**Comment by wookayin** [comment]
Hmm I find that using tox makes running tests on *multiple* python versions easy. Now I'm happy with keeping it (made some improvements, #536) The only issue I have now is that tox unfortunately silences and suppresses all the pytest output despite `-s --verbose` option, but this is something we can figure out in the future.

**Pull request merged**